File: src/main/kotlin/br/com/fiap/techfood/adapters/outbound/repository/entities/OrderEntity.java
package br.com.fiap.techfood.adapters.outbound.repository.entities;

import br.com.fiap.techfood.application.core.domains.enums.OrderStatusEnum;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@Entity
@Table(name = "TB_ORDERS")
public class OrderEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Integer status;
    private Boolean isAnonymous;

    @OneToMany(mappedBy = "id.order")
    private List<OrderItemEntity> items;

    @ManyToOne
    @JoinColumn(name = "client_id")
    private ClientEntity client;

    @Column(nullable = false)
    private LocalDateTime creationDate;

    @Column(nullable = false)
    private LocalDateTime lastUpdateDate;

    @PrePersist
    private void beforePersist() {
        this.creationDate = LocalDateTime.now();
        this.lastUpdateDate = LocalDateTime.now();
    }

    @PreUpdate
    private void beforeUpdate() {
        this.lastUpdateDate = LocalDateTime.now();
    }

    private OrderStatusEnum getStatus() {
        return OrderStatusEnum.toEnum(this.status);
    }

    private void setStatus(OrderStatusEnum status) {
        this.status = status.getCode();
    }


}

File: src/main/kotlin/br/com/fiap/techfood/adapters/outbound/repository/entities/OrderEntity.kt
package br.com.fiap.techfood.adapters.outbound.repository.entities

import br.com.fiap.techfood.application.core.domains.enums.OrderStatusEnum
import jakarta.persistence.*
import lombok.Data
import java.time.LocalDateTime
import java.util.*

@Data
@Entity
@Table(name = "TB_ORDERS")
class OrderEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null

    @Column(nullable = false)
    var name: String? = null

    @Column(nullable = false)
    private var status: Int? = null

    public fun getStatus(): OrderStatusEnum? {
        return OrderStatusEnum.toEnum(this.status)
    }

    public fun setStatus(status: OrderStatusEnum) {
        this.status = status.code
    }

    var isAnonymous: Boolean? = null

    @OneToMany(mappedBy = "id.order")
    var items: List<OrderItemEntity> = listOf();

    @ManyToOne
    @JoinColumn(name = "client_id", foreignKey = ForeignKey(name = "fk_order_client"))
    var client: ClientEntity? = null

    @Column(nullable = false)
    var creationDate: LocalDateTime? = null

    @Column(nullable = false)
    var lastUpdateDate: LocalDateTime? = null

    @PrePersist
    private fun beforePersist() {
        this.creationDate = LocalDateTime.now()
        this.lastUpdateDate = LocalDateTime.now()
    }

    @PreUpdate
    private fun beforeUpdate() {
        this.lastUpdateDate = LocalDateTime.now()
    }
}
